sympy/solvers/solveset.py

```diff
@@ -3196,7 +3444,12 @@ def _solve_using_known_values(result, solver):
                         # and add this new solution
                         if soln_imageset:
                             # replace all lambda variables with 0.
-                            imgst = soln_imageset[sol]
+                            try:
+                                imgst = soln_imageset[sol]
+                            # imageset(lambda(),) with 0 returns keyerror
+                            except KeyError:
+                                continue
```

Comment:
I don't understand why this is here. Why wouldn't `sol` be in `soln_imageset`?

Reply:
Actually here the evaluation of `imageset` is done if other than `Imageset` enter here then it shows error...
So, either `KeyError ` should be there or `if soln_imageset and sol.has(Dummy):` is to be used (at line 3303 and 3445).

Reply:
> if other than Imageset enter here then it shows error

Since both `non-lambert` and `lambert` solutions are input to this logic, so both of them entering into this loop hence for non-lambert (where `imageset` is not present) should be diverted ....

Reply:
> Please suggest any of the two here ...